File: 118-pascals-triangle/pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        ansList = []
        prev = []
        i=0
        while(i<numRows):
            if(i==0):
                prev = [1]
                ansList.append(prev)
            else:
                curr = [1]
                j = 1
                while(j<i):
                    curr.append(prev[j-1]+prev[j])
                    j+=1
                curr.append(1)
                ansList.append(curr)
                prev = curr
            i+=1
        return ansList
        
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> prev = new ArrayList<>();
        for(int i=0;i<numRows;i++){
            List<Integer> curr = new ArrayList<>();
            if(i==0){
                curr.add(1);
            }else {
                curr.add(1);
                for(int j=1;j<prev.size();j++){
                    curr.add(prev.get(j)+prev.get(j-1));
                }
                curr.add(1);
            }
            prev = curr;
            ans.add(curr);
            curr = new ArrayList<>();
        }   
        return ans;
    }
}